### PR TITLE
grpc: 1.38.1 -> 1.39.0

### PR DESCRIPTION
**Pull request by marsam**
###### Motivation for this change
https://github.com/grpc/grpc/releases/tag/v1.39.0

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by marsam** [comment]
@GrahamcOfBorg build murmur bear

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 131021` at cae1f984 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>12 packages marked as broken and skipped:</summary>
  <ul>
    <li>python38Packages.ignite</li>
    <li>python38Packages.pytorchWithCuda</li>
    <li>python38Packages.spacy</li>
    <li>python38Packages.spacy-transformers</li>
    <li>python38Packages.tensorflowWithCuda</li>
    <li>python38Packages.textacy</li>
    <li>python39Packages.ignite</li>
    <li>python39Packages.pytorchWithCuda</li>
    <li>python39Packages.spacy</li>
    <li>python39Packages.spacy-transformers</li>
    <li>python39Packages.tensorflowWithCuda</li>
    <li>python39Packages.textacy</li>
  </ul>
</details>
<details>
  <summary>9 packages failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/875fec45a905c11fc9f7f52edf921109">python38Packages.cirq-google</a></li>
    <li><a href="https://gist.github.com/eb7111cd3940632d845933b8e4d05308">python38Packages.google-cloud-error-reporting</a></li>
    <li><a href="https://gist.github.com/27cb25c0a489a6df88f561a2be7f43db">python38Packages.google-cloud-logging</a></li>
    <li><a href="https://gist.github.com/d65acd1cef708a5e5ab2ca1762505653">python38Packages.pathy</a></li>
    <li><a href="https://gist.github.com/618c8525a6a22c38b763a6f51100f850">python38Packages.skein</a></li>
    <li><a href="https://gist.github.com/01ebd6631f3e432c80336492f4cfa94f">python39Packages.cirq-google</a></li>
    <li><a href="https://gist.github.com/b74b629c621be69ff376b080f49fd4c6">python39Packages.google-cloud-error-reporting</a></li>
    <li><a href="https://gist.github.com/36c8e601aecf3da36d454184004619a7">python39Packages.google-cloud-logging</a></li>
    <li><a href="https://gist.github.com/730ba10aff736f77b258c326d4a7402f">python39Packages.skein</a></li>
  </ul>
</details>
<details>
  <summary>63 packages skipped due to time constraints:</summary>
  <ul>
    <li>home-assistant</li>
    <li>kmymoney</li>
    <li>python38Packages.arviz</li>
    <li>python38Packages.baselines</li>
    <li>python38Packages.cirq</li>
    <li>python38Packages.dm-sonnet</li>
    <li>python38Packages.edward</li>
    <li>python38Packages.graph_nets</li>
    <li>python38Packages.mask-rcnn</li>
    <li>python38Packages.optuna</li>
    <li>...</li>
  </ul>
</details>
<details>
  <summary>133 packages built successfully:</summary>
  <ul>
    <li>beancount (python39Packages.beancount)</li>
    <li>bear</li>
    <li>charge-lnd</li>
    <li>cinnamon.warpinator</li>
    <li>deja-dup</li>
    <li>duplicity</li>
    <li>duply</li>
    <li>dvc-with-remotes</li>
    <li>fava</li>
    <li>gcalcli</li>
    <li>git-annex-remote-googledrive</li>
    <li>grpc</li>
    <li>ledger2beancount</li>
    <li>lieer</li>
    <li>lndmanage</li>
    <li>murmur</li>
    <li>opensnitch-ui</li>
    <li>python38Packages.beancount</li>
    <li>python38Packages.beancount_docverif</li>
    <li>python38Packages.chirpstack-api</li>
    <li>python38Packages.drivelib</li>
    <li>python38Packages.gensim</li>
    <li>python38Packages.goobook</li>
    <li>python38Packages.google-api-core</li>
    <li>python38Packages.google-api-python-client</li>
    <li>python38Packages.google-cloud-access-context-manager</li>
    <li>python38Packages.google-cloud-appengine-logging</li>
    <li>python38Packages.google-cloud-asset</li>
    <li>python38Packages.google-cloud-automl</li>
    <li>python38Packages.google-cloud-bigquery</li>
    <li>python38Packages.google-cloud-bigquery-datatransfer</li>
    <li>python38Packages.google-cloud-bigquery-logging</li>
    <li>python38Packages.google-cloud-bigtable</li>
    <li>python38Packages.google-cloud-container</li>
    <li>python38Packages.google-cloud-core</li>
    <li>python38Packages.google-cloud-dataproc</li>
    <li>python38Packages.google-cloud-datastore</li>
    <li>python38Packages.google-cloud-dlp</li>
    <li>python38Packages.google-cloud-dns</li>
    <li>python38Packages.google-cloud-firestore</li>
    <li>python38Packages.google-cloud-iam</li>
    <li>python38Packages.google-cloud-iam-logging</li>
    <li>python38Packages.google-cloud-iot</li>
    <li>python38Packages.google-cloud-kms</li>
    <li>python38Packages.google-cloud-language</li>
    <li>python38Packages.google-cloud-monitoring</li>
    <li>python38Packages.google-cloud-org-policy</li>
    <li>python38Packages.google-cloud-os-config</li>
    <li>python38Packages.google-cloud-pubsub</li>
    <li>python38Packages.google-cloud-redis</li>
    <li>python38Packages.google-cloud-resource-manager</li>
    <li>python38Packages.google-cloud-runtimeconfig</li>
    <li>python38Packages.google-cloud-secret-manager</li>
    <li>python38Packages.google-cloud-securitycenter</li>
    <li>python38Packages.google-cloud-spanner</li>
    <li>python38Packages.google-cloud-speech</li>
    <li>python38Packages.google-cloud-storage</li>
    <li>python38Packages.google-cloud-tasks</li>
    <li>python38Packages.google-cloud-texttospeech</li>
    <li>python38Packages.google-cloud-trace</li>
    <li>python38Packages.google-cloud-translate</li>
    <li>python38Packages.google-cloud-videointelligence</li>
    <li>python38Packages.google-cloud-vision</li>
    <li>python38Packages.google-cloud-websecurityscanner</li>
    <li>python38Packages.googleapis-common-protos</li>
    <li>python38Packages.grpc_google_iam_v1</li>
    <li>python38Packages.grpcio</li>
    <li>python38Packages.grpcio-gcp</li>
    <li>python38Packages.grpcio-tools</li>
    <li>python38Packages.oauthenticator</li>
    <li>python38Packages.proto-plus</li>
    <li>python38Packages.pydrive</li>
    <li>python38Packages.smart-open</li>
    <li>python38Packages.weboob</li>
    <li>python38Packages.woob</li>
    <li>python39Packages.beancount_docverif</li>
    <li>python39Packages.chirpstack-api</li>
    <li>python39Packages.drivelib</li>
    <li>python39Packages.gensim</li>
    <li>python39Packages.goobook</li>
    <li>python39Packages.google-api-core</li>
    <li>python39Packages.google-api-python-client</li>
    <li>python39Packages.google-cloud-access-context-manager</li>
    <li>python39Packages.google-cloud-appengine-logging</li>
    <li>python39Packages.google-cloud-asset</li>
    <li>python39Packages.google-cloud-automl</li>
    <li>python39Packages.google-cloud-bigquery</li>
    <li>python39Packages.google-cloud-bigquery-datatransfer</li>
    <li>python39Packages.google-cloud-bigquery-logging</li>
    <li>python39Packages.google-cloud-bigtable</li>
    <li>python39Packages.google-cloud-container</li>
    <li>python39Packages.google-cloud-core</li>
    <li>python39Packages.google-cloud-dataproc</li>
    <li>python39Packages.google-cloud-datastore</li>
    <li>python39Packages.google-cloud-dlp</li>
    <li>python39Packages.google-cloud-dns</li>
    <li>python39Packages.google-cloud-firestore</li>
    <li>python39Packages.google-cloud-iam</li>
    <li>python39Packages.google-cloud-iam-logging</li>
    <li>python39Packages.google-cloud-iot</li>
    <li>python39Packages.google-cloud-kms</li>
    <li>python39Packages.google-cloud-language</li>
    <li>python39Packages.google-cloud-monitoring</li>
    <li>python39Packages.google-cloud-org-policy</li>
    <li>python39Packages.google-cloud-os-config</li>
    <li>python39Packages.google-cloud-pubsub</li>
    <li>python39Packages.google-cloud-redis</li>
    <li>python39Packages.google-cloud-resource-manager</li>
    <li>python39Packages.google-cloud-runtimeconfig</li>
    <li>python39Packages.google-cloud-secret-manager</li>
    <li>python39Packages.google-cloud-securitycenter</li>
    <li>python39Packages.google-cloud-spanner</li>
    <li>python39Packages.google-cloud-speech</li>
    <li>python39Packages.google-cloud-storage</li>
    <li>python39Packages.google-cloud-tasks</li>
    <li>python39Packages.google-cloud-texttospeech</li>
    <li>python39Packages.google-cloud-trace</li>
    <li>python39Packages.google-cloud-translate</li>
    <li>python39Packages.google-cloud-videointelligence</li>
    <li>python39Packages.google-cloud-vision</li>
    <li>python39Packages.google-cloud-websecurityscanner</li>
    <li>python39Packages.googleapis-common-protos</li>
    <li>python39Packages.grpc_google_iam_v1</li>
    <li>python39Packages.grpcio</li>
    <li>python39Packages.grpcio-gcp</li>
    <li>python39Packages.grpcio-tools</li>
    <li>python39Packages.oauthenticator</li>
    <li>python39Packages.proto-plus</li>
    <li>python39Packages.pydrive</li>
    <li>python39Packages.smart-open</li>
    <li>python39Packages.weboob</li>
    <li>python39Packages.woob</li>
    <li>snakemake</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md">maintainers-missing</a>

Package does not have a maintainer. Consider adding yourself?

Near pkgs/development/python-modules/grpcio-tools/default.nix:27:5:
```
   |
27 |     maintainers = with maintainers; [ ];
   |     ^
```
</li>
  </ul>
</details>

Note that build failures [may predate](https://github.com/nix-community/hydra-check) this PR, and could be nondeterministic or hardware dependent.
Please exercise your independent judgement. Does something look off? Please file an [issue](https://github.com/rmcgibbo/r-rmcgibbo/issues) or reach out on IRC.
